File: stepik-java-api/src/main/java/org/stepik/api/queries/progresses/StepikProgressesGetQuery.java
package org.stepik.api.queries.progresses;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.actions.StepikAbstractAction;
import org.stepik.api.objects.progresses.Progresses;
import org.stepik.api.queries.StepikAbstractGetQuery;
import org.stepik.api.urls.Urls;

/**
 * @author meanmail
 */
public class StepikProgressesGetQuery extends StepikAbstractGetQuery<StepikProgressesGetQuery, Progresses> {
    public StepikProgressesGetQuery(StepikAbstractAction stepikAction) {
        super(stepikAction, Progresses.class);
    }

    @NotNull
    @Override
    protected String getUrl() {
        return Urls.PROGRESSES;
    }

    @Override
    protected boolean isCacheEnabled() {
        return false;
    }

    @NotNull
    public StepikProgressesGetQuery page(int page) {
        addParam("page", page);
        return this;
    }
}

File: stepik-java-api/src/main/kotlin/org/stepik/api/queries/progresses/StepikProgressesGetQuery.kt
package org.stepik.api.queries.progresses

import org.stepik.api.actions.StepikAbstractAction
import org.stepik.api.objects.progresses.Progresses
import org.stepik.api.queries.StepikAbstractGetQuery

class StepikProgressesGetQuery(stepikAction: StepikAbstractAction) :
        StepikAbstractGetQuery<StepikProgressesGetQuery, Progresses>(stepikAction, Progresses::class.java) {
    
    override fun getUrl(): String {
        return "${stepikAction.stepikApiClient.host}/api/progresses"
    }
    
    override fun isCacheEnabled(): Boolean {
        return false
    }
    
    fun page(page: Int): StepikProgressesGetQuery {
        addParam("page", page)
        return this
    }
}
